src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/TransPrimitiveClass.java

```diff
@@ -389,14 +389,20 @@ private MethodSymbol getPrimitiveObjectFactory(MethodSymbol init) {
         if (factory != null)
             return factory;
 
-        MethodType factoryType = new MethodType(init.erasure(types).getParameterTypes(),
+        MethodType factoryType = new MethodType(init.type.getParameterTypes(),
                                                 init.owner.type.asValueType(),
                                                 init.type.getThrownTypes(),
```

Comment:
I am lost and need a bit of help here. What exactly is the change and why is it needed ?

Reply:
This method makes a primitive object factory from the parsed constructor. In the current version, the parameter types are erased and their annotations, and original method signature, are erased from the original. This line preserves this information, so the generated classfile is correct.

See next comment.